tensorflow/compiler/xla/service/gpu/gpu_fusible.cc

```diff
@@ -506,5 +504,24 @@ HloInstruction::FusionKind ChooseFusionKind(const HloInstruction& /*producer*/,
                                   : HloInstruction::FusionKind::kLoop;
 }
 
+bool IsConsumerTheOnlyNonRootUser(const HloInstruction& instr,
+                                  const HloInstruction& consumer) {
+  return absl::c_all_of(instr.users(), [&](const HloInstruction* user) {
+    if (user->opcode() == HloOpcode::kGetTupleElement) {
+      // Skip GTE.
+      return IsConsumerTheOnlyNonRootUser(*user, consumer);
+    } else if (user == &consumer) {
+      // `user` is `consumer`.
+      return true;
+    } else if (user == user->parent()->root_instruction()) {
+      // Consumed by ROOT is always fine, since it is impossible to create
+      // cycles through ROOT.
```

Comment:
Not clear why we care about cycles here. Please clarify the comment.

Reply:
Revised. We simply should not have these comments here. They are confusing.